pkg/domain/attributor.go

```diff
@@ -0,0 +1,41 @@
+package domain
+
+import (
+	"context"
+
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/nexpose"
+)
+
+// AssetAttributor receives a NexposeAssetVulnerabilities instance, queries one
+// or more asset inventory systems, and returns a NexposeAttributedAssetVulnerabilities
+// instance annotated with the business context for the asset at scan time.
+type AssetAttributor interface {
+	Attribute(ctx context.Context, asset NexposeAssetVulnerabilities) (NexposeAttributedAssetVulnerabilities, error)
+}
+
+// NexposeAssetVulnerabilities is a Nexpose asset response payload appended
+// with assetVulnerabilityDetails
+type NexposeAssetVulnerabilities struct {
+	Asset           *nexpose.Asset
```

Comment:
Assuming the payload matches the doc I don't think this will work as it will expect a field `Asset` on the incoming payload. I guess we should decide on whether an API call constitutes a new field or if we start with an asset and keep appending?

Reply:
Great catch! This was actually me just forgetting how embedding works in Go, but after thinking through it, I do think we should start with an asset and keep appending, because that will be the pattern we'll use for vulnerability hydration as well, meaning that, e.g., each solution will be nested underneath its corresponding vulnerability, not placed at the root of the payload.

Reply:
Why is this a pointer?